### PR TITLE
Fix Clang warnings

### PR DESCRIPTION
**Pull request by heplesser**
This PR fixes a large number of warnings emitted by Apple Clang. Almost all of these warnings were due to explicit template class instantiations for Datum types, which could be removed by moving minimal methods from `_impl.h` files to `.h` files. This allowed removal of several `_impl.h` and `.cc` files.

### CODE REVIEW

**Review by jougs — APPROVED** [review]
Thanks for this cleanup. I did not follow each move individually, but trust the fact that the CI is happy.

File: nestkernel/parameter.cpp [hunk 1]
```diff
@@ -27,12 +27,9 @@
 #include "spatial.h"
 #include "vp_manager_impl.h"
 
-// includes from sli
-#include "sharedptrdatum.h"
```
Comment (hakonsbm) [inline nestkernel/parameter.cpp:30]: I think a single blank line here is sufficient.
```suggestion
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]
Looks good. I just have one very minor formatting suggestion.

**Review by hakonsbm — APPROVED** [review]
Thanks!

**Pull request merged**